Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeSwitcher.cpp

```diff
@@ -176,55 +174,54 @@ namespace AzToolsFramework::ComponentModeFramework
 
         ComponentData newComponentData = ComponentData(pairId);
 
+        auto componentDataIt = AZStd::find_if(
```

Comment:
nit: This could be marked `const`

(I'd also maybe suggest moving this back to be in the if initializer but if this version is easier to read then feel free to leave as-is)